parquet-avro/src/main/java/org/apache/parquet/avro/AvroRecordConverter.java

```diff
@@ -930,6 +930,7 @@ public void end() {
   static boolean isElementType(Type repeatedType, Schema elementSchema) {
     if (repeatedType.isPrimitive()
         || repeatedType.asGroupType().getFieldCount() > 1
+        || repeatedType.getName().equals("array")
```

Comment:
This now matches the List [logical-type spec](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#backward-compatibility-rules):

> If the repeated field is a group with one field and is named either array or uses the LIST-annotated group's name with _tuple appended then the repeated type is the element type and elements are required.

as well as the equivalent isElementType check in AvroSchemaConverter: https://github.com/apache/parquet-mr/blob/61bdf44/parquet-avro/src/main/java/org/apache/parquet/avro/AvroSchemaConverter.java#L575-L584

Note: We don't have to add another clause to check for the `_tuple` style naming scheme, as that's already covered by line 934: `repeatedType.asGroupType().getType(0).isRepetition(REPEATED))`.

Reply:
```
1. If the repeated field is not a group, then its type is the element type and elements are required.
2. If the repeated field is a group with multiple fields, then its type is the element type and elements are required.
3. If the repeated field is a group with one field and is named either array or uses the LIST-annotated group's name with _tuple appended then the repeated type is the element type and elements are required.
4. Otherwise, the repeated field's type is the element type with the repeated field's repetition.
```

`repeatedType.getName().equals("array")` is partially for rule 3, and `repeatedType.asGroupType().getType(0).isRepetition(REPEATED))` is for rule 4. But rule 3 also states the _tuple style naming scheme should come with `the repeated field is a group with one field`, isn't it? If the repeated field is a group with multiple fields with `_tuple` appended, is it still a valid element type?

Reply:
oh interesting, so a schema like this?

```proto
message MySchema {
  optional group list_field (LIST) {
    repeated group list_field_tuple_1 (LIST) {
      required int32 int_field_1;
    }
    repeated group list_field_tuple_2 (LIST) {
      required int32 int_field_2
    }
  }
}
```

I think luckily, it's impossible to create this in Avro. I tried running it through AvroSchemaConverter, and it threw an exception:

```java
 > MessageType parquetSchema = MessageTypeParser.parseMessageType(
        "message MySchema {\n"
        + "  optional group list_field (LIST) {\n"
        + "    repeated group list_field_tuple_1 (LIST) {\n"
        + "      required int32 int_field_1;\n"
        + "    }\n"
        + "    repeated group list_field_tuple_2 (LIST) {\n"
        + "      required int32 int_field_2;\n"
        + "    }\n"
        + "  }\n"
        + "}");

> new AvroSchemaConverter().convert(parquetSchema));
java.lang.UnsupportedOperationException: Invalid list type optional group list_field (LIST) {
  repeated group list_field_tuple_1 (LIST) {
    required int32 int_field_1;
  }
  repeated group list_field_tuple_2 (LIST) {
    required int32 int_field_2;
  }
}
```